### PR TITLE
Minitest/RefuteRespondTo

### PR DESCRIPTION
**Pull request by herwinw**
Similar to `Minitest/AssertRespondTo`, includes a lot of copy-paste. The style guide was present.

-----------------

Before submitting the PR make sure the following are checked:

* [ ] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [ ] Feature branch is up-to-date with `master` (if not - rebase it).
* [ ] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop-minitest/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop-minitest/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [ ] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [ ] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -56,6 +56,12 @@ Minitest/RefuteIncludes:
   Enabled: true
   VersionAdded: '0.3'
 
+Minitest/RefuteRespondTo:
+  Description: 'This cop enforces the test to use `refute_respond_to(object, :some_method) of `refute(object.respond_to?(:some_method))`.'
```
Comment (abhaynikam) [inline config/default.yml]:
```suggestion
  Description: 'This cop enforces the test to use `refute_respond_to(object, :some_method) over `refute(object.respond_to?(:some_method))`.'
```

**Comment by abhaynikam** [comment]
@herwinw : you need to add a changelog entry for this change. Rest LGTM 🎉

**Comment by herwinw** [comment]
Changelog has been added as well.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## master (unreleased)
 
+### New features
+
+* [#29](https://github.com/rubocop-hq/rubocop-minitest/pull/29): Add new `Minitest/RefuteRespondTo` cop.  ([@herwinw][])
```
Comment (abhaynikam) [inline CHANGELOG.md]: @herwinw : Can you add your GitHub profile URL here(https://github.com/rubocop-hq/rubocop-minitest/pull/29/files#diff-4ac32a78649ca5bdd8e0ba38b7006a1eL44)? Your username should be clickable in the changelog.

Comment (herwinw) [reply CHANGELOG.md]: Fixed that as well.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**